common/meta/types/src/user_grant.rs

```diff
@@ -205,40 +205,27 @@ impl UserGrantSet {
         &self.grants
     }
 
-    pub fn verify_global_privilege(
+    pub fn verify_privilege(
```

Comment:
common/meta/types mainly for type definiton.
Maybe better to put the `verify_privilege` in other place?

Reply:
after adding roles, we might need a seperate PrivilegeManager to cache and verify the user privileges (it needs cache all the roles' privileges, and generate an effective grant set with the privileges merged from the roles), maybe the PrivilegeManager could be the better place to verify the privilege, and leaving the types user_grant mod as plain data objects.  🤔

this refactor requires some work, maybe we can make this refactor in a stand-alone PR? q.q

Reply:
That's OK